core/src/main/scala/cats/instances/vector.scala

```diff
@@ -130,6 +130,13 @@ trait VectorInstances extends cats.kernel.instances.VectorInstances {
       final override def traverse[G[_], A, B](fa: Vector[A])(f: A => G[B])(implicit G: Applicative[G]): G[Vector[B]] =
         G.map(Chain.traverseViaChain(fa)(f))(_.toVector)
 
+      final override def updated_[A, B >: A](fa: Vector[A], idx: Long, b: B): Option[Vector[B]] =
```

Comment:
We can add a similar override to `NonEmptyVector`.

Reply:
Added one in https://github.com/typelevel/cats/pull/4248/commits/f9a00e686d82ec144c01ab6dd835fc148c401e59.